hazelcast-client/src/main/java/com/hazelcast/client/spi/ProxyManager.java

```diff
@@ -319,6 +320,8 @@ private void initializeWithRetry(ClientProxy clientProxy) throws Exception {
                 }
             }
         }
+        throw new OperationTimeoutException("Initializing  " + clientProxy.getServiceName() + ":"
+                + clientProxy.getName() + " timed out");
```

Comment:
Can you append the actual time consumed and the configured timeout value to the log so that it will help troubleshoot if any problem?